### PR TITLE
move several methods out of line

### PR DESCRIPTION
**Pull request by neheb**
Avoids -Wweak-vtables. Small size reduction.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by whyman** [comment]
Thanks, can you add the -Wweak-vtables to CMake too?

File: src/content/scripting/import_script.cc [hunk 1]
```diff
@@ -49,6 +49,11 @@ ImportScript::ImportScript(std::shared_ptr<ContentManager> content,
     }
 }
 
+script_class_t ImportScript::whoami()
+{
+    return S_PLAYLIST;
```
Comment (KarlStraussberger) [inline src/content/scripting/import_script.cc:54]: Must be `S_IMPORT` here

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]

**Comment by neheb** [comment]
> Thanks, can you add the -Wweak-vtables to CMake too?

Not all are fixed or should be fixed.

**Comment by whyman** [comment]
> > Thanks, can you add the -Wweak-vtables to CMake too?
> 
> Not all are fixed or should be fixed.

I don't understand why this would be?

**Comment by neheb** [comment]
Basically, it would require placing out of line definitions for virtual functions. This is somewhat problematic for defaulted destructors. Destructors should always be defaulted in their definitions. Not doing so just increases size for no good reason.

**Comment by neheb** [comment]
updated slightly.

**Pull request merged**